### ISSUE TITLE
Cannot integrate latest versions of Firebase, GoogleDataTransport and nanopb with CocoaPods

### ISSUE THREAD

**Issue by dmazzoni** [firebase/firebase-ios-sdk#12477]
### Description

I am updating a Firebase integration to version `10.22.0`. To include all privacy manifests for transitive dependencies, I have also bumped GoogleDataTransport to `9.40.0` and nanopb to `2.30910.0`.

With the following setup, `pod install` fails with the following error:

```
[!] CocoaPods could not find compatible versions for pod "nanopb":
  In Podfile:
    FirebaseCrashlytics (= 10.22.0) was resolved to 10.22.0, which depends on
      nanopb (< 2.30911.0, >= 2.30908.0)

    GoogleDataTransport (= 9.4.0) was resolved to 9.4.0, which depends on
      nanopb (< 2.30910.0, >= 2.30908.0)

    nanopb (= 2.30910.0)
```

The root cause seems to be the fact that GoogleDataTransport's [published Podspec](https://github.com/CocoaPods/Specs/blob/07ba88919ba6193a2d284c73ce904321734d2e9b/Specs/0/6/a/GoogleDataTransport/9.4.0/GoogleDataTransport.podspec.json#L53) for version `9.40.0` requires nanopb to have a version lower than `2.30910.0`, preventing the integration of the nanopb version which includes the privacy manifest.

### Reproducing the issue

The issue can be reproduced by cloning this [sample repo](https://github.com/dmazzoni/FirebaseTest) and running a `pod install`.

### Firebase SDK Version

10.22.0

### Xcode Version

15.0.1

### Installation Method

CocoaPods

### Firebase Product(s)

Crashlytics

### Targeted Platforms

iOS

### If using CocoaPods, the project's Podfile.lock

No Podfile.lock is produced, as `pod install` fails.

**Comment by ncooke3** [comment]
Apologies, seems like there was an inconsistency in what was published. Working on a fix...

**Comment by ncooke3** [comment]
I've just released GoogleDataTransport 9.4.1. Please run pod update to get this latest fix. You should see GoogleDataTransport bump to 9.4.1.
